### PR TITLE
Features/rules configs management client

### PR DESCRIPTION
**Pull request by caldwell0414**
### Changes

Added a new Client implementation to the ManagementApi named RulesConfigClient.  This client will make calls to the /rules-configs endpoint in the Auth0 Management API (v2).  It includes a PUT and a DELETE call.  It does not include any GET requests because they are not supported on that endpoint for security purposes.  Because the implementor cannot determine if a variable already exists, we have named the PUT implementation CreateOrUpdateAsync().

We followed the pattern provided for other clients in the library and included the great cleanup work done by @frederikprijck.  His commit is included in our PR, so you may want to accept his first.

### References

See Issue #551 for more details.

### Testing

Our PR includes integration tests to exercise the new client.

- [ ] This change adds unit test coverage

- [x] This change adds integration test coverage

- [x] This change has been tested on the latest version of the platform/language or why not

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)

- [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)

- [ ] All existing and new tests complete without errors
       NOTE: some tests require configuration that we don't have correct on our test host.  please let us know if your testing runs into any critical failures.  We did not modify any existing code and so do not expect any side effects.

### CODE REVIEW

**Comment by frederikprijck** [comment]
Thanks for this!

Let me first merge https://github.com/auth0/auth0.net/pull/550, then we can rebase your branch and get it reviewed and merged as well.

**Comment by caldwell0414** [comment]
Thanks very much.  Please let me know if you need anything.  I’ve been using Auth0 for a long time and I’m happy to be able to contribute.

jc

**Comment by frederikprijck** [comment]
Hey @caldwell0414 ,

Would you be able to resolve the conflicts? I just merged the other PR so you should be able to drop the cleanup commit, and merge in master.

Thanks

File: src/Auth0.ManagementApi/Clients/RulesConfigClient.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        /// <param name="defaultHeaders">Dictionary containing default headers included with every request this client makes.</param>
+        public RulesConfigClient(IManagementConnection connection, Uri baseUri, IDictionary<string, string> defaultHeaders)
+            : base(connection, baseUri, defaultHeaders)
+        {
+        }
+
+        /// <summary>
+        /// Creates or updates a rules config variable according to the request.
+        /// </summary>
+        /// <param name="request">The <see cref="RuleCreateRequest" /> containing the details of the rule to create.</param>
```
Comment (frederikprijck) [inline src/Auth0.ManagementApi/Clients/RulesConfigClient.cs:29]:
```suggestion
        /// <param name="request">The <see cref="RulesConfigCreateOrUpdateRequest" /> containing the details of the rule to create or update.</param>
```

**Review by frederikprijck — CHANGES_REQUESTED** [review]

**Review by frederikprijck — APPROVED** [review]

**Comment by frederikprijck** [comment]
I update the base here, thanks! 🚀

**Pull request merged**